src/main/java/gov/usgs/ngwmn/logic/WaterLevelStatistics.java

```diff
@@ -134,6 +94,11 @@ protected void removeMostRecentProvisional(List<WLSample> samples, List<WLSample
 	}
 	
 	// TODO business rule for any one value error? continue without value or error entire statistics calculation?
+	@Override
+	public JsonData calculate(Specifier spec, Reader xmlData) throws Exception {
+		List<WLSample> samples = WLSample.extractSamples(xmlData, spec.getAgencyCd(), spec.getSiteNo(), spec.getElevation());
+		return calculate(spec, samples);
+	}
```

Comment:
This method was originally only used in the ngwmn-cache app. However, I have to code in the statistics project as well. I have data from the site that has issues and it felt best to test the actual data rather than trying to construct a fictitious set. So, this creates a method to read from XML streams that could have been here during the first extraction from the cache code.